src/planwise/client/scenarios/edit.cljs

```diff
@@ -26,31 +26,29 @@
                :accept-fn  #(dispatch [:scenarios/accept-rename-dialog])
                :cancel-fn  #(dispatch [:scenarios/cancel-rename-dialog])}))))
 
-(defn input
-  [{:keys [value onchange-path]}]
-  [common2/text-field {:type "number"
-                       :on-change  #(dispatch [:scenarios/save-key onchange-path (-> % .-target .-value int)])
-                       :value value}])
-
 (defn changeset-dialog-content
-  [{:keys [investment capacity]}]
-  [:div
-   [:h2 "Investment"]
-   [input {:value (or investment "")
-           :onchange-path [:changeset-dialog :investment]}]
-   [:h2 "Capacity"]
-   [input {:value (or capacity "")
-           :onchange-path [:changeset-dialog :capacity]}]])
+  [{:keys [investment capacity]} scenario-investment project-budget]
+  (let [custom-text-prop (when (< project-budget (+ scenario-investment investment)) " invalid-input")]
```

Comment:
Intent here is to compute if the budget is valid or not, which is kind of lost in translation. So, I'd compute that (as a boolean) and use it later to output the "extra class".